server/buildbuddy_server/buildbuddy_server.go

```diff
@@ -1187,16 +1188,10 @@ func (s *BuildBuddyServer) serveUsingParams(w http.ResponseWriter, r *http.Reque
 			http.Error(w, err.Error(), code)
 			return
 		}
-		// Attempt to fall back gracefully if the specified file is not found and
-		// a fallback is specified.
-		if fallback, parseErr := url.Parse(params.Get("with_fallback")); parseErr != nil {
-			http.Error(w, parseErr.Error(), http.StatusBadRequest)
-		} else if fallback.RawQuery == "" {
-			http.Error(w, fmt.Sprintf("Fallback without query is invalid: %v", fallback), http.StatusBadRequest)
-		} else {
-			q := fallback.Query()
-			q.Del("with_fallback") // Only fall back once.
-			s.serveUsingParams(w, r, q)
+		// Attempt to fall back gracefully if the specified file is not found
```

Comment:
I think we should only do this fallback in the "bytestream_url" case above, since it looks like in the "artifact" case it's just a retry and not a fallback, if I'm reading it right?